### PR TITLE
Fixes and renaming in dps_guidance.py

### PR DESCRIPTION
**Pull request by CharlelieLrt**
# PhysicsNeMo Pull Request

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/diffusion/guidance/dps_guidance.py [hunk 1]
```diff
@@ -390,6 +388,14 @@ def __call__(
             Guided score of same shape :math:`(B, *)` as ``x``. Computed as the
             sum of the unconditional score and all guidance terms.
         """
+        if not torch.compiler.is_compiling() and torch.is_inference_mode_enabled():
+            raise RuntimeError(
+                "DPSScorePredictor requires autograd but torch inference mode "
+                "is enabled. Wrap the calling code with "
+                "'with torch.inference_mode(False):' or 'with torch.no_grad():' "
+                "instead."
+            )
```
[the remark this answers is not in the document]
Comment (CharlelieLrt) [reply physicsnemo/diffusion/guidance/dps_guidance.py:397]: I think it's just a matter of grammar, and the former sentence sounds correct to me

File: physicsnemo/diffusion/guidance/dps_guidance.py [hunk 2]
```diff
@@ -459,9 +465,10 @@ def norm(
         ``(y_pred, y_true)`` and returns a scalar loss per batch element of
         shape :math:`(B,)`.
     gamma : float, default=0.0
-        SDA scaling parameter. When ``gamma > 0``, applies SDA correction
-        that accounts for the variance of the x0 estimate. Set to ``0`` for
-        classical DPS without SDA scaling.
+        SDA covariance scaling factor :math:`\Gamma`. When ``gamma > 0``,
+        applies SDA correction that accounts for the covariance of the
+        :math:`\hat{\mathbf{x}}_0` estimate at different noise levels. Set
+        to ``0`` for classical DPS without SDA scaling.
```
Comment (NickGeneva) [inline physicsnemo/diffusion/guidance/dps_guidance.py:471]: Is this technically true, isnt it if \Gamma and std_y are both 0 it will fall back to regular DPS?

File: physicsnemo/diffusion/guidance/dps_guidance.py [hunk 3]
```diff
@@ -991,7 +1006,14 @@ def __call__(
         Tensor
             Likelihood score guidance term of same shape as ``x``.
         """
-        # Ensure stored tensors match x's dtype and device
+        if not torch.compiler.is_compiling() and torch.is_inference_mode_enabled():
+            raise RuntimeError(
+                "DataConsistencyDPSGuidance requires autograd but torch "
+                "inference mode is enabled. Wrap the calling code with "
+                "'with torch.inference_mode(False):' or 'with torch.no_grad():' "
```
Comment (NickGeneva) [inline physicsnemo/diffusion/guidance/dps_guidance.py:1013]: So the `torch.enable_grad()` will work inside a `torch.no_grad()` context? Just confirming.

File: physicsnemo/diffusion/guidance/dps_guidance.py
Comment (NickGeneva) [inline physicsnemo/diffusion/guidance/dps_guidance.py:1046]: Is it worth doing a `grad_x.detach()` here?

File: physicsnemo/diffusion/guidance/dps_guidance.py [hunk 4]
```diff
@@ -768,9 +782,10 @@ def norm(
         ``(mask * x_0, mask * y)`` and returns a scalar loss per batch element
         of shape :math:`(B,)`.
     gamma : float, default=0.0
-        SDA scaling parameter. When ``gamma > 0``, applies SDA correction
-        that accounts for the variance of the x0 estimate. Set to ``0`` for
-        classical DPS without SDA scaling.
+        SDA covariance scaling factor :math:`\Gamma`. When ``gamma > 0``,
+        applies SDA correction that accounts for the covariance of the
+        :math:`\hat{\mathbf{x}}_0` estimate at different noise levels. Set
+        to ``0`` for classical DPS without SDA scaling.
```
Comment (NickGeneva) [inline physicsnemo/diffusion/guidance/dps_guidance.py:788]: similar comment as above

File: physicsnemo/diffusion/guidance/dps_guidance.py
Comment (NickGeneva) [inline physicsnemo/diffusion/guidance/dps_guidance.py:408]: Is it worth to detach guidance_sum here?

and or also self.x0_to_score_fn(x_0.detach(), x.detach(), t.detach())

Idk just a thought.

Comment (NickGeneva) [inline physicsnemo/diffusion/guidance/dps_guidance.py:1]: Do unit tests need updating are are there none atm

File: physicsnemo/diffusion/guidance/dps_guidance.py (hunk 2 above)
Comment (CharlelieLrt) [reply physicsnemo/diffusion/guidance/dps_guidance.py:471]: No, if both Gamma and std_y are 0, you get an infinitely strong guidance.
Original DPS is usually formulated as: `rho * ||observation_model(x0) - y||^2`, where `rho` is the guidance strength that is a constant (independent of the noise schedule `sigma(t)`, `alpha(t)`). So Gamma=0 gives you the same with `rho = 1/std_y**2`

File: physicsnemo/diffusion/guidance/dps_guidance.py (hunk 3 above)
Comment (CharlelieLrt) [reply physicsnemo/diffusion/guidance/dps_guidance.py:1013]: I confirm this works

File: physicsnemo/diffusion/guidance/dps_guidance.py
Comment (CharlelieLrt) [reply physicsnemo/diffusion/guidance/dps_guidance.py:1]: There are none at the moment 😅

Comment (CharlelieLrt) [reply physicsnemo/diffusion/guidance/dps_guidance.py:408]: - By default `retain_graph` and `create_graph` are both False, so detaching guidance_sum would have nearly no effect at all.
- x is already detached
- detaching t would have very little effect
- detaching x0 might be more beneficial, but that would also close to door to applications that need to backpropagate through the entire sampler (or its right hand side). For example, I am not sure how ditsillation works with guidance, but I am excluding the possibility that it needs to backprop through the guidance. Enabling such applications was actually the point of adding the arguments `retain_graph` and `create_graph`.

Comment (CharlelieLrt) [reply physicsnemo/diffusion/guidance/dps_guidance.py:1046]: By default the arguments `retain_graph` and `create_graph` are False, so this grad_x,.detach would have very little effect (the graph has already been mostly destroyed after calling `torch.autograd.grad`

File: physicsnemo/diffusion/guidance/dps_guidance.py (hunk 4 above)
Comment (CharlelieLrt) [reply physicsnemo/diffusion/guidance/dps_guidance.py:788]: replied above
